### PR TITLE
Prompt when logging in from LTI

### LINKED ISSUE #2212: Explicitly ask for course registration when deep linking from LTI
If a user lands on a course page directly from an LTI redirect, we should add a modal that explicitly asks to register for the course.

This is a request from i-Learn and will also improve the Ufora experience.

### PR DESCRIPTION
**Pull request by niknetniko**
This pull request adds a prompt when logging in from LTI. Technically, opening a link via LTI is a `LtiResourceLinkRequest`, so we use that to detect LTI and extract some data (we use the name of the learning platform if available, otherwise "een andere leeromgeving").

The dialog is shown on:

- The course page (also when linking to series)
- The series page (e.g. a hidden series)
- The activity page

Currently the dialog looks like this:
image

Closes #2212.

Tasks:

- [x] Deploy & test on naos

### CODE REVIEW

File: app/controllers/concerns/set_lti_message.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative '../../../lib/LTI/jwk.rb'
```
Comment (thepieterdc) [inline app/controllers/concerns/set_lti_message.rb:1]: Good abstraction :+1:

File: app/views/courses/_not_a_member_dialog.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title"><%= t 'courses.registration.lti_title' %></h4>
+      </div>
+      <div class="modal-body">
+        <div class="row">
+          <div class="col-md-2">
+            <%= image_tag "idp/#{@provider.institution.logo}", class: "img-responsive" %>
+          </div>
+          <div class="col-md-10">
+            <p><%= t('courses.registration.lti', lms: @lti_message.platform_name || t('courses.registration.lms_default')) %></p>
```
Comment (thepieterdc) [inline app/views/courses/_not_a_member_dialog.html.erb:14]: Maybe we can use `provider.institution.name` because then we have control over the name and we are sure that it always exists.

File: config/locales/views/courses/en.yml [hunk 3]
```diff
@@ -144,6 +144,9 @@ en:
       not_logged_in: You need to be logged in to register
       key_mismatch: "The key didn't match"
       registration_closed: "Registration closed."
+      lti_title: Welkom
```
Comment (thepieterdc) [inline config/locales/views/courses/en.yml:147]: Dutch

File: app/views/courses/_not_a_member_dialog.html.erb (hunk 2 above)
Comment (niknetniko) [reply app/views/courses/_not_a_member_dialog.html.erb:14]: I'm not sure if this won't be confusing for students with the current wording. If we want to use this, we should maybe reword it to something like "We hebben gemerkt dat je een link hebt gevolgd uit een leeromgeving van $institution".

**Review by charvp — APPROVED** [review]

File: app/views/courses/_not_a_member_dialog.html.erb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title"><%= t 'courses.registration.lti_title' %></h4>
+      </div>
+      <div class="modal-body">
+        <div class="row">
+          <div class="col-md-2">
+            <%= image_tag "idp/#{@provider.institution.logo}", class: "img-responsive" %>
```
Comment (bmesuere) [inline app/views/courses/_not_a_member_dialog.html.erb:11]: I would not use the image because not all institutions have one (only the ones with a button on the home page).

File: config/locales/views/courses/nl.yml [hunk 5]
```diff
@@ -140,6 +140,9 @@ nl:
       not_logged_in: Je moet ingelogd zijn om je te registreren
       key_mismatch: De sleutel kwam niet overeen
       registration_closed: "Registratie gesloten."
+      lti_title: Welkom
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:143]:
```suggestion
      lti_title: Welkom op Dodona!
```

File: config/locales/views/courses/en.yml [hunk 6]
```diff
@@ -144,6 +144,9 @@ en:
       not_logged_in: You need to be logged in to register
       key_mismatch: "The key didn't match"
       registration_closed: "Registration closed."
+      lti_title: Welcome
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:147]:
```suggestion
      lti_title: Welcome to Dodona!
```

File: config/locales/views/courses/nl.yml [hunk 7]
```diff
@@ -140,6 +140,9 @@ nl:
...
+      lti: "We merken dat je een link volgde uit %{lms}, maar dat je nog niet bent ingeschreven voor deze cursus. Klik op registreren om dat wel te doen zodat de lesgever je kan opvolgen."
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:144]:
```suggestion
      lti: "Je volgde een link uit %{lms} naar deze pagina. Om je ingediende oplossingen correct bij te houden is het nodig om je op Dodona te registreren voor deze cursus. Dat kan eenvoudig door hieronder op registreren te klikken, of later vanaf de cursuspagina."
```

File: config/locales/views/courses/en.yml [hunk 8]
```diff
@@ -144,6 +144,9 @@ en:
...
+      lti: "We've noticed you followed a link from %{lms}, but you're not registered for this course yet. Click on register to enable the course administrator to follow your progress."
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:148]:
```suggestion
      lti: "You followed a link from %{lms} to this page. To correctly store your submissions, you should register for this course on Dodona. You can do this easily by clicking the "register" button below, or later from the course page."
```

**Review by bmesuere — CHANGES_REQUESTED** [review]
👍 Code looks good, a few small UI/text suggestions. Nice work!

**Review by bmesuere — APPROVED** [review]
